Do not Save Panels if a TrainIcon is visible

In versions 4.9.2, 4.9.1, 4.8, and perhaps earlier, a Layout Editor panel cannot be loaded if it contains a TrainIcon object.  

There are two ERROR messages triggered by  single TrainIcon object in the following log.  It appears as if the first ERROR might not be critical but the second ERROR stops the panel from loading.

Thus far, my only known fix is to edit the xml file and delete all trainIcon lines.

This is not a high priority problem, as it only seems to hit me when editing the Operations Routes data for icon locations and save intermediate files to double check the results.  Double checking is a long burned in habit, but still not done enough.  
We do not perform a Save Panels at the club when ending an Ops Session.  

'''
2017-08-13 20:12:42,829 util.Log4JUtil                        INFO  - ****** JMRI log ******* [main]
2017-08-13 20:12:42,892 util.Log4JUtil                        INFO  - This log is appended to file: C:\Users\cliffaa-rr\JMRI\log\messages.log [main]
2017-08-13 20:12:42,892 util.Log4JUtil                        INFO  - This log is stored in file: C:\Users\cliffaa-rr\JMRI\log\session.log [main]
2017-08-13 20:12:42,923 apps.Apps                             INFO  - PanelPro version 4.9.2+Rff24066 starts under Java 1.8.0_144 on Windows 10 amd64 v10.0 at Sun Aug 13 20:12:42 PDT 2017 [main]
2017-08-13 20:12:46,096 apps.Apps                             INFO  - Starting with profile Layout_Ops_Test.3f053837 [main]
2017-08-13 20:12:46,908 node.NodeIdentity                     INFO  - Using jmri-F81654D51042-3f053837 as the JMRI Node identity [main]
2017-08-13 20:12:47,111 managers.JmriUserPreferencesManager   ERROR - Unable to retrieve property "windowRectangle2D" for window "jmri.jmrit.display.layoutEditor.LayoutEditor:Test TrainIcon" [main]
2017-08-13 20:12:49,268 hexfile.LnHexFilePort                 INFO  - LocoNet Simulator Started [Thread-8]
2017-08-13 20:12:50,064 loconet.LnTrafficController           DEBUG - notify of incoming LocoNet packet: B0 78 27 10 [AWT-EventQueue-0]
2017-08-13 20:12:50,080 loconet.LnTrafficController           DEBUG - notify of incoming LocoNet packet: BB 00 00 44 [AWT-EventQueue-0]
2017-08-13 20:12:50,080 loconet.LnTrafficController           DEBUG - notify of incoming LocoNet packet: B0 79 27 11 [AWT-EventQueue-0]
2017-08-13 20:12:50,736 apps.PerformFileModel                 INFO  - Loading file C:\Users\cliffaa-rr\JMRI\Config Profiles\Layout_Ops_Test\panels\Test TrainIcon 1.xml [main]
2017-08-13 20:12:50,877 loconet.LnTrafficController           DEBUG - notify of incoming LocoNet packet: B0 7A 27 12 [AWT-EventQueue-0]
2017-08-13 20:12:52,877 configurexml.ErrorHandler             ERROR - Parse error while parsing file /C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml Exception: org.jdom2.input.JDOMParseException: Error on line 49: cvc-complex-type.2.4.a: Invalid content was found starting with element 'trainicon'. One of '{positionablelabel, positionablepoint, turnouticon, sensoricon, memoryicon, memoryInputIcon, signalheadicon, signalmasticon, reportericon, layoutturnout, layoutSlip, tracksegment, fastclock, multisensoricon, levelxing, layoutturntable, linkinglabel, locoicon, BlockContentsIcon, LightIcon}' is expected.
See http://jmri.org/help/en/package/jmri/configurexml/ErrorHandler.shtml for possibly more information. [AWT-EventQueue-0]
org.jdom2.input.JDOMParseException: Error on line 49: cvc-complex-type.2.4.a: Invalid content was found starting with element 'trainicon'. One of '{positionablelabel, positionablepoint, turnouticon, sensoricon, memoryicon, memoryInputIcon, signalheadicon, signalmasticon, reportericon, layoutturnout, layoutSlip, tracksegment, fastclock, multisensoricon, levelxing, layoutturntable, linkinglabel, locoicon, BlockContentsIcon, LightIcon}' is expected.
	at org.jdom2.input.sax.SAXBuilderEngine.build(SAXBuilderEngine.java:232)
	at org.jdom2.input.sax.SAXBuilderEngine.build(SAXBuilderEngine.java:253)
	at org.jdom2.input.SAXBuilder.build(SAXBuilder.java:1091)
	at jmri.jmrit.XmlFile.getRoot(XmlFile.java:197)
	at jmri.jmrit.XmlFile.rootFromURL(XmlFile.java:182)
	at jmri.configurexml.ConfigXmlManager.loadOnSwingThread(ConfigXmlManager.java:648)
	at jmri.configurexml.ConfigXmlManager.lambda$1(ConfigXmlManager.java:615)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)
Caused by: org.xml.sax.SAXParseException; lineNumber: 49; columnNumber: 354; cvc-complex-type.2.4.a: Invalid content was found starting with element 'trainicon'. One of '{positionablelabel, positionablepoint, turnouticon, sensoricon, memoryicon, memoryInputIcon, signalheadicon, signalmasticon, reportericon, layoutturnout, layoutSlip, tracksegment, fastclock, multisensoricon, levelxing, layoutturntable, linkinglabel, locoicon, BlockContentsIcon, LightIcon}' is expected.
	at org.apache.xerces.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
	at org.apache.xerces.util.ErrorHandlerWrapper.error(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.XMLErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.xs.XMLSchemaValidator$XSIErrorReporter.reportError(Unknown Source)
	at org.apache.xerces.impl.xs.XMLSchemaValidator.reportSchemaError(Unknown Source)
	at org.apache.xerces.impl.xs.XMLSchemaValidator.handleStartElement(Unknown Source)
	at org.apache.xerces.impl.xs.XMLSchemaValidator.startElement(Unknown Source)
	at org.apache.xerces.xinclude.XIncludeHandler.startElement(Unknown Source)
	at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source)
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source)
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source)
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source)
	at org.jdom2.input.sax.SAXBuilderEngine.build(SAXBuilderEngine.java:217)
	... 20 more
'''

--- a/java/test/jmri/jmrit/display/configurexml/verify/TestTrainIcon.xml
+++ b/java/test/jmri/jmrit/display/configurexml/verify/TestTrainIcon.xml
@@ -1,0 +1,168 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet href="/xml/XSLT/panelfile-2-9-6.xsl" type="text/xsl"?>
+<layout-config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://jmri.org/xml/schema/layout-2-9-6.xsd">
+  <jmriversion>
+    <major>4</major>
+    <minor>9</minor>
+    <test>2</test>
+    <modifier />
+  </jmriversion>
+  <sensors class="jmri.jmrix.internal.configurexml.InternalSensorManagerXml">
+    <defaultInitialState>unknown</defaultInitialState>
+    <sensor inverted="false">
+      <systemName>ISCLOCKRUNNING</systemName>
+    </sensor>
+  </sensors>
+  <memories class="jmri.managers.configurexml.DefaultMemoryManagerXml">
+    <memory value="10:51 PM">
+      <systemName>IMCURRENTTIME</systemName>
+    </memory>
+    <memory value="1.0">
+      <systemName>IMRATEFACTOR</systemName>
+    </memory>
+  </memories>
+  <signalmasts class="jmri.managers.configurexml.DefaultSignalMastManagerXml" />
+  <signalgroups class="jmri.managers.configurexml.DefaultSignalGroupManagerXml" />
+  <oblocks class="jmri.jmrit.logix.configurexml.OBlockManagerXml" />
+  <warrants class="jmri.jmrit.logix.configurexml.WarrantManagerXml" />
+  <signalmastlogics class="jmri.managers.configurexml.DefaultSignalMastLogicManagerXml">
+    <logicDelay>500</logicDelay>
+  </signalmastlogics>
+  <timebase class="jmri.jmrit.simpleclock.configurexml.SimpleTimebaseXml" time="Wed Aug 02 11:26:40 PDT 2017" rate="1.0" run="yes" master="yes" sync="no" correct="no" display="no" startstopped="no" startsettime="no" startclockoption="0" showbutton="no" />
+  <entryexitpairs class="jmri.jmrit.signalling.configurexml.EntryExitPairsXml" />
+  <LayoutEditor class="jmri.jmrit.display.layoutEditor.configurexml.LayoutEditorXml" name="Test TrainIcon" x="0" y="0" height="960" width="1900" windowheight="302" windowwidth="1392" panelheight="960" panelwidth="1900" sliders="no" scrollable="none" editable="no" positionable="yes" controlling="yes" animating="yes" showhelpbar="no" drawgrid="yes" snaponadd="no" snaponmove="no" antialiasing="no" turnoutcircles="no" tooltipsnotedit="no" tooltipsinedit="yes" mainlinetrackwidth="4" xscale="1.0" yscale="1.0" sidetrackwidth="2" defaulttrackcolor="black" defaultoccupiedtrackcolor="red" defaultalternativetrackcolor="white" defaulttextcolor="black" turnoutcirclecolor="black" turnoutcirclesize="4" turnoutdrawunselectedleg="yes" turnoutbx="20.0" turnoutcx="20.0" turnoutwid="10.0" xoverlong="30.0" xoverhwid="10.0" xovershort="10.0" autoblkgenerate="no" gridSize="10" gridSize2nd="10" openDispatcher="no" useDirectTurnoutControl="no">
+    <positionablelabel x="100" y="50" level="4" forcecontroloff="false" hidden="no" positionable="true" showtooltip="false" editable="false" text="A" fontname="Tahoma Bold" size="28" style="1" red="0" green="0" blue="0" hasBackground="no" justification="centre" class="jmri.jmrit.display.configurexml.PositionableLabelXml">
+      <tooltip>Text Label</tooltip>
+    </positionablelabel>
+    <positionablelabel x="300" y="50" level="4" forcecontroloff="false" hidden="no" positionable="true" showtooltip="false" editable="false" text="B" fontname="Tahoma Bold" size="28" style="1" red="0" green="0" blue="0" hasBackground="no" justification="centre" class="jmri.jmrit.display.configurexml.PositionableLabelXml">
+      <tooltip>Text Label</tooltip>
+    </positionablelabel>
+    <positionablelabel x="500" y="50" level="4" forcecontroloff="false" hidden="no" positionable="true" showtooltip="false" editable="false" text="C" fontname="Tahoma Bold" size="28" style="1" red="0" green="0" blue="0" hasBackground="no" justification="centre" class="jmri.jmrit.display.configurexml.PositionableLabelXml">
+      <tooltip>Text Label</tooltip>
+    </positionablelabel>
+    <positionablelabel x="700" y="50" level="4" forcecontroloff="false" hidden="no" positionable="true" showtooltip="false" editable="false" text="D" fontname="Tahoma Bold" size="28" style="1" red="0" green="0" blue="0" hasBackground="no" justification="centre" class="jmri.jmrit.display.configurexml.PositionableLabelXml">
+      <tooltip>Text Label</tooltip>
+    </positionablelabel>
+    <positionablelabel x="900" y="50" level="4" forcecontroloff="false" hidden="no" positionable="true" showtooltip="false" editable="false" text="E" fontname="Tahoma Bold" size="28" style="1" red="0" green="0" blue="0" hasBackground="no" justification="centre" class="jmri.jmrit.display.configurexml.PositionableLabelXml">
+      <tooltip>Text Label</tooltip>
+    </positionablelabel>
+    <trainicon train="1" trainName="100" x="105" y="70" level="10" forcecontroloff="false" hidden="no" positionable="true" showtooltip="true" editable="false" text="100" size="16" style="0" red="0" green="0" blue="0" hasBackground="no" justification="centre" icon="yes" dockX="0" dockY="0" class="jmri.jmrit.operations.trains.configurexml.TrainIconXml">
+      <tooltip>A-E at Ayee next Beee</tooltip>
+      <icon url="program:resources/icons/markers/loco-green.gif" degrees="0" scale="1.0">
+        <rotation>0</rotation>
+      </icon>
+    </trainicon>
+    <tracksegment ident="T1" connect1name="A1" type1="1" connect2name="A2" type2="1" dashed="no" mainline="no" hidden="no" arc="no" class="jmri.jmrit.display.layoutEditor.configurexml.TrackSegmentXml" />
+    <tracksegment ident="T2" connect1name="A2" type1="1" connect2name="A3" type2="1" dashed="no" mainline="no" hidden="no" arc="no" class="jmri.jmrit.display.layoutEditor.configurexml.TrackSegmentXml" />
+    <tracksegment ident="T3" connect1name="A3" type1="1" connect2name="A4" type2="1" dashed="no" mainline="no" hidden="no" arc="no" class="jmri.jmrit.display.layoutEditor.configurexml.TrackSegmentXml" />
+    <tracksegment ident="T4" connect1name="A4" type1="1" connect2name="A5" type2="1" dashed="no" mainline="no" hidden="no" arc="no" class="jmri.jmrit.display.layoutEditor.configurexml.TrackSegmentXml" />
+    <positionablepoint ident="A1" type="1" x="100.0" y="100.0" connect1name="T1" class="jmri.jmrit.display.layoutEditor.configurexml.PositionablePointXml" />
+    <positionablepoint ident="A2" type="1" x="300.0" y="100.0" connect1name="T1" connect2name="T2" class="jmri.jmrit.display.layoutEditor.configurexml.PositionablePointXml" />
+    <positionablepoint ident="A3" type="1" x="500.0" y="100.0" connect1name="T2" connect2name="T3" class="jmri.jmrit.display.layoutEditor.configurexml.PositionablePointXml" />
+    <positionablepoint ident="A4" type="1" x="700.0" y="100.0" connect1name="T3" connect2name="T4" class="jmri.jmrit.display.layoutEditor.configurexml.PositionablePointXml" />
+    <positionablepoint ident="A5" type="1" x="900.0" y="100.0" connect1name="T4" class="jmri.jmrit.display.layoutEditor.configurexml.PositionablePointXml" />
+  </LayoutEditor>
+  <filehistory>
+    <operation>
+      <type>app</type>
+      <date>Sun Aug 13 22:50:55 PDT 2017</date>
+      <filename>JMRI program</filename>
+    </operation>
+    <operation>
+      <type>Load OK</type>
+      <date>Sun Aug 13 22:51:06 PDT 2017</date>
+      <filename>/C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml</filename>
+      <filehistory>
+        <operation>
+          <type>app</type>
+          <date>Sun Aug 13 20:11:08 PDT 2017</date>
+          <filename>JMRI program</filename>
+        </operation>
+        <operation>
+          <type>Load OK</type>
+          <date>Sun Aug 13 20:11:20 PDT 2017</date>
+          <filename>/C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml</filename>
+          <filehistory>
+            <operation>
+              <type>app</type>
+              <date>Wed Aug 02 21:33:18 PDT 2017</date>
+              <filename>JMRI program</filename>
+            </operation>
+            <operation>
+              <type>Load OK</type>
+              <date>Wed Aug 02 21:33:21 PDT 2017</date>
+              <filename>/C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml</filename>
+              <filehistory>
+                <operation>
+                  <type>app</type>
+                  <date>Wed Aug 02 18:42:04 PDT 2017</date>
+                  <filename>JMRI program</filename>
+                </operation>
+                <operation>
+                  <type>Load OK</type>
+                  <date>Wed Aug 02 18:42:10 PDT 2017</date>
+                  <filename>/C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml</filename>
+                  <filehistory>
+                    <operation>
+                      <type>app</type>
+                      <date>Wed Aug 02 12:56:41 PDT 2017</date>
+                      <filename>JMRI program</filename>
+                    </operation>
+                    <operation>
+                      <type>Load OK</type>
+                      <date>Wed Aug 02 12:56:53 PDT 2017</date>
+                      <filename>/C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml</filename>
+                      <filehistory>
+                        <operation>
+                          <type>app</type>
+                          <date>Wed Aug 02 11:51:42 PDT 2017</date>
+                          <filename>JMRI program</filename>
+                        </operation>
+                        <operation>
+                          <type>Load OK</type>
+                          <date>Wed Aug 02 11:51:53 PDT 2017</date>
+                          <filename>/C:/Users/cliffaa-rr/JMRI/Config%20Profiles/Layout_Ops_Test/panels/Test%20TrainIcon%201.xml</filename>
+                        </operation>
+                        <operation>
+                          <type>Store</type>
+                          <date>Wed Aug 02 11:52:58 PDT 2017</date>
+                          <filename />
+                        </operation>
+                      </filehistory>
+                    </operation>
+                    <operation>
+                      <type>Store</type>
+                      <date>Wed Aug 02 13:54:43 PDT 2017</date>
+                      <filename />
+                    </operation>
+                  </filehistory>
+                </operation>
+                <operation>
+                  <type>Store</type>
+                  <date>Wed Aug 02 18:51:13 PDT 2017</date>
+                  <filename />
+                </operation>
+              </filehistory>
+            </operation>
+            <operation>
+              <type>Store</type>
+              <date>Wed Aug 02 21:46:07 PDT 2017</date>
+              <filename />
+            </operation>
+          </filehistory>
+        </operation>
+        <operation>
+          <type>Store</type>
+          <date>Sun Aug 13 20:12:32 PDT 2017</date>
+          <filename />
+        </operation>
+      </filehistory>
+    </operation>
+    <operation>
+      <type>Store</type>
+      <date>Sun Aug 13 22:51:59 PDT 2017</date>
+      <filename />
+    </operation>
+  </filehistory>
+  <!--Written by JMRI version 4.9.2+Rff24066 on Sun Aug 13 22:51:59 PDT 2017-->
+</layout-config>

--- a/scripts/HOWTO-distribution.md
+++ b/scripts/HOWTO-distribution.md
@@ -60,7 +60,7 @@ If you're attempting to perform this on MS Windows, refer to the MS Windows note
 
 - Update this note by (details in the update-HOWTO.sh comments):
 ```
-  ./scripts/update-HOWTO.sh 4.9.1 4.9.2 4.9.3
+  ./scripts/update-HOWTO.sh 4.9.2 4.9.3 4.9.4
 ```
 (and then manually update that line above to be last version release, this version being made today, next version to be made later; i.e. when starting to do *.4, the arguments are *.3 *.4 *.5)
 
@@ -167,13 +167,13 @@ We roll some general code maintenance items into the release process.  They can 
 ```    
         cd (local web copy)/releasenotes
         git pull 
-        cp jmri4.7.7.shtml jmri4.9.1.shtml
+        cp jmri4.9.2.shtml jmri4.9.3.shtml
         (edit the new release note accordingly)
             change numbers throughout
             move new warnings to old
             remove old-version change notes
-        git add jmri4.9.1.shtml
-        git commit -m"start new 4.9.1 release note" jmri4.9.1.shtml
+        git add jmri4.9.2.shtml
+        git commit -m"start new 4.9.3 release note" jmri4.9.3.shtml
         PR-and-merge (or direct push) and pull back.
         cd (local JMRI copy)
 ```
@@ -208,13 +208,13 @@ where the date at the end should be the date (and optionally time) of the last r
 - Put a comment in the release GitHub item saying the branch exists, and all future changes should be documented in the new release note
 
 ```
-The release-4.9.1 branch has been created. 
+The release-4.9.2 branch has been created. 
 
-From now on, please document your changes in the [jmri4.9.2.shtml](https://github.com/JMRI/website/blob/master/releasenotes/jmri4.9.2.shtml) release note file.
+From now on, please document your changes in the [jmri4.9.3.shtml](https://github.com/JMRI/website/blob/master/releasenotes/jmri4.9.3.shtml) release note file.
 
-Maintainers, please set the 4.9.2 milestone on pulls from now on, as that will be the next test release from the HEAD of the master branch.
+Maintainers, please set the 4.9.3 milestone on pulls from now on, as that will be the next test release from the HEAD of the master branch.
 
-Jenkins will be creating files shortly at the [CI server](http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.1/)
+Jenkins will be creating files shortly at the [CI server](http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.2/)
 ````
 
 ================================================================================
@@ -252,7 +252,7 @@ If you're building locally:
 - Get the release in your local work directory
 
 ```
-    git checkout release-4.9.1
+    git checkout release-4.9.2
 ```
 
 - edit release.properties to say release.official=true (last line)
@@ -287,11 +287,11 @@ If you're building locally:
 
 - Change the release note to point to the just-built files (in CI or where you put them), commit, wait (or force via ["Build Now"](http://jmri.tagadab.com/jenkins/job/Web%20Site/job/Website%20from%20JMRI%20GitHub%20website%20repository/) update). Confirm visible on web.
 
-- Announce the file set via email to jmri-developers@lists.sf.net with a subject line "First 4.9.1 files available":
+- Announce the file set via email to jmri-developers@lists.sf.net with a subject line "First 4.9.2 files available":
 
-First JMRI 4.9.1 files are available in the usual way at:
+First JMRI 4.9.2 files are available in the usual way at:
 
-http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.1
+http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.2
 
 Feedback appreciated. I would like to release this later today or tomorrow morning. 
 
@@ -302,13 +302,15 @@ Feedback appreciated. I would like to release this later today or tomorrow morni
 
 If anybody wants to add a change from here on in, they should
 
-- commit it to a branch of their own, and push as needed to get it to their GitHub fork.
+- Ideally, start the work on either the release-4.9.2 branch (if working after that was started) or on a branch-from-master that's _before_ the release-4.9.2 branch was created.  That way, the change can be cleanly included in the release branch, and also directly onto master.
+
+- Commit their changes to that branch, and push as needed to get it to their GitHub fork.
 
 - On the GitHub web site, make _two_ pull requests:  
 
    - One to master, as usual
    
-   - One to the release branch e.g. "release-4.9.1".  The comment on this PR should explain why this should be included instead of waiting for the next release.
+   - One to the release branch e.g. "release-4.9.2".  The comment on this PR should explain why this should be included instead of waiting for the next release.
    
   Merging the PR to the master makes those changes available on further developments forever; the one on the release, if accepted, includes the change and kicks off new runs of the various CI and build jobs.
 
@@ -339,13 +341,13 @@ This step uploads the Linux, Mac OS X and Windows files to the SourceForge file 
 
 (replace "user" below with your SourceForge.net user name; must have SSH keys for SourceForge.net set up)
 
- - (The "./testrelease 4.9.1" local script on shell.sf.net does the following steps, except for the edit, of course)
+ - (The "./testrelease 4.9.2" local script on shell.sf.net does the following steps, except for the edit, of course)
 ```
     ssh user,jmri@shell.sf.net create
     ssh user,jmri@shell.sf.net
-    curl -o release.zip "http://jmri.tagadab.com/jenkins/job/Test%20Releases/job/4.9.1/ws/dist/release/*zip*/release.zip"
+    curl -o release.zip "http://jmri.tagadab.com/jenkins/job/Test%20Releases/job/4.9.2/ws/dist/release/*zip*/release.zip"
         (use the following instead if building on second Jenkins server)
-    curl -o release.zip "http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.1/ws/dist/release/*zip*/release.zip"
+    curl -o release.zip "http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.2/ws/dist/release/*zip*/release.zip"
     rm release/JMRI*
     unzip release.zip
     cd release
@@ -394,27 +396,28 @@ Note: Unlike releasing files to SourceForge, once a GitHub Release is created it
 
 - Fill out form:
 
-   - "tag version field" gets v4.9.1 (e.g. leading lower-case "v")
-   - @ branch: select the release-4.9.1 release branch
+   - "tag version field" gets v4.9.2 (e.g. leading lower-case "v")
+   - @ branch: select the release-4.9.2 release branch
 ```
-"Release title" field gets "Test/Prod Release 4.9.1"
+"Release title" field gets "Test/Prod Release 4.9.2"
 ```
    - Description content (the testrelease script above proposed this!):
 ```    
-[Release notes](http://jmri.org/releasenotes/jmri4.9.1.shtml)
+[Release notes](http://jmri.org/releasenotes/jmri4.9.2.shtml)
 
 Checksums:
 
 File | SHA256 checksum
 ---|---
-[JMRI.4.9.1.R583bddd.dmg](https://github.com/JMRI/JMRI/releases/download/v4.9.1/JMRI.4.9.1.R583bddd.dmg) | 688b6da8a2e66ad77ef6499d34381e82eb4eba88ad37477de6b24a11906cdf21
-[JMRI.4.9.1.R583bddd.exe](https://github.com/JMRI/JMRI/releases/download/v4.9.1/JMRI.4.9.1.R583bddd.exe) | 071d54f1ce97b8bf701567a0c44f63104462f7966bcc5319112f320cf133da8c
-[JMRI.4.9.1.R583bddd.tgz](https://github.com/JMRI/JMRI/releases/download/v4.9.1/JMRI.4.9.1.R583bddd.tgz) | 804a8b715cb11983cad287fb58356644f36482ec2f02563f83ecbaff63f3f07e
+[JMRI.4.9.2.Rff24066.dmg](https://github.com/JMRI/JMRI/releases/download/v4.9.2/JMRI.4.9.2.Rff24066.dmg) | 61cc6eb66a6a7600376990e7181ae119fef33846824073f86483a18f5e7ad725
+[JMRI.4.9.2.Rff24066.exe](https://github.com/JMRI/JMRI/releases/download/v4.9.2/JMRI.4.9.2.Rff24066.exe) | 2b21fa42fd1979cc65450a9e8ed97c1274e1a6c5ac2f68955a130f33383e0f98
+[JMRI.4.9.2.Rff24066.tgz](https://github.com/JMRI/JMRI/releases/download/v4.9.2/JMRI.4.9.2.Rff24066.tgz) | 794fd80964d710aee2a962d2597b8a7e9329e238e16399a7f8cb6e64595b1c98
+
 ```
 
 - Attach files by selecting them or dragging them in (you might have to have downloaded them above via e.g. a separate 
 ```
-curl -o release.zip "http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.1/lastSuccessfulBuild/artifact/dist/release/*zip*/release.zip"" 
+curl -o release.zip "http://jmri.tagadab.com/jenkins/job/TestReleases/job/4.9.2/lastSuccessfulBuild/artifact/dist/release/*zip*/release.zip"" 
 ```
 and expansion; it's slow to upload from a typical home machine, though, so wish we had a way to cross-load from somewhere fast - if release.zip is still on SF.net, you can do
 ```
@@ -428,9 +431,9 @@ Note there's a little progress bar that has to go across & "Uploading your relea
 Alternatively, if you have shell access to the Jenkins server, you perhaps can upload directly from there, once the initial draft release has been created (this hasn't been tested):
 
 ```
-github-release upload -s {github_secret} -u JMRI -r JMRI -t v4.9.1 -n "JMRI.4.9.1+Rd144052.dmg" -f /var/lib/jenkins/jobs/TestReleases/jobs/4.9.1/workspace/dist/release/JMRI.4.9.1+Rd144052.dmg 
-github-release upload -s {github_secret} -u JMRI -r JMRI -t v4.9.1 -n "JMRI.4.9.1+Rd144052.exe" -f /var/lib/jenkins/jobs/TestReleases/jobs/4.7.5/workspace/dist/release/JMRI.4.9.1+Rd144052.exe 
-github-release upload -s {github_secret} -u JMRI -r JMRI -t v4.9.1 -n "JMRI.4.9.1+Rd144052.tgz" -f /var/lib/jenkins/jobs/TestReleases/jobs/4.7.5/workspace/dist/release/JMRI.4.9.1+Rd144052.tgz 
+github-release upload -s {github_secret} -u JMRI -r JMRI -t v4.9.2 -n "JMRI.4.9.2+Rd144052.dmg" -f /var/lib/jenkins/jobs/TestReleases/jobs/4.9.2/workspace/dist/release/JMRI.4.9.2+Rd144052.dmg 
+github-release upload -s {github_secret} -u JMRI -r JMRI -t v4.9.2 -n "JMRI.4.9.2+Rd144052.exe" -f /var/lib/jenkins/jobs/TestReleases/jobs/4.7.5/workspace/dist/release/JMRI.4.9.2+Rd144052.exe 
+github-release upload -s {github_secret} -u JMRI -r JMRI -t v4.9.2 -n "JMRI.4.9.2+Rd144052.tgz" -f /var/lib/jenkins/jobs/TestReleases/jobs/4.7.5/workspace/dist/release/JMRI.4.9.2+Rd144052.tgz 
 ```
     
 - Click "Publish Release"
@@ -448,10 +451,10 @@ git fetch
 git checkout master
 git pull
 git checkout -b temp-master
-git merge (release-n.n.n)
+git merge origin/release-4.9.2
 ```
 
-Note that you're testing the merge of the release branch back onto master.  This should show no changes, with the possible exception of some auto-generated files:
+Note that you're testing the merge of the release branch back onto master.  This should report "Already up-to-date.", i.e. no changes, with the possible exception of some auto-generated files:
 ```
 xml/decoderIndex.xml
 help/en/webindex.shtml
@@ -464,6 +467,8 @@ If there are any changes in other files, do both of:
    - Make sure they get moved back to the master branch
 
    - Figure out what went wrong and fix it in these instructions
+   
+- You can delete that temp-master local branch now
 
 Lastly, if this release is one of the special series at the end of a development cycle that leads to a test release, create the next release branch now.  Those test releases are made cumulatively from each other, rather than each from master. We start the process now so that people can open pull requests for it, and discuss whether changes should be included.
 
@@ -480,7 +485,7 @@ git push github
 
 - Create the next [GitHub Issue](https://github.com/JMRI/JMRI/issues) to hold discussion with conventional title "Create release-n.n.n+1". Add the next release milestone (created above) to it.
 
-- Confirm that the tag for the current release (release-4.9.1) is in place, then manually delete the current release branch via the [GitHub UI](https://github.com/JMRI/JMRI/branches).
+- Confirm that the tag for the current release (release-4.9.2) is in place, then manually delete the current release branch via the [GitHub UI](https://github.com/JMRI/JMRI/branches).
 
 - Go to the GitHub PR and Issues [labels list](https://github.com/JMRI/JMRI/labels) and remove any "afterNextTestRelease" (and "afterNextProductionRelease" if appropriate) labels from done items
 
@@ -510,26 +515,26 @@ If you don't, a bunch of Windows users are likely to whine
 
 - Mail announcement to jmriusers@yahoogroups.com
 
-    Subject is "Test version 4.9.1 of JMRI/DecoderPro is available for download" or "JMRI 4.8 is available for download"
+    Subject is "Test version 4.9.2 of JMRI/DecoderPro is available for download" or "JMRI 4.8 is available for download"
 
     Content:
     
-Test version 4.9.1 of JMRI/DecoderPro is available for download.
+Test version 4.9.2 of JMRI/DecoderPro is available for download.
 
-This is the next in a series of test releases that will culminate in a production release, hopefully in June 2017.
+This is the next in a series of test releases that will culminate in a production release, hopefully in December 2017.
 
 There have been a lot of updates in this version, so it should be considered experimental.
 
-If you use JMRI on Linux or Mac and are updating from an earlier version, there’s a necessary migration step. (Not needed on Windows) Please see the release note for details: <http://jmri.org/releasenotes/jmri4.9.1.shtml#migration>
+If you use JMRI on Linux or Mac and are updating from an earlier version, there’s a necessary migration step. (Not needed on Windows) Please see the release note for details: <http://jmri.org/releasenotes/jmri4.9.2.shtml#migration>
 
-For more information on the issues, new features and bug fixes in 4.9.1 please see the release note:
-<http://jmri.org/releasenotes/jmri4.9.1.shtml>
+For more information on the issues, new features and bug fixes in 4.9.2 please see the release note:
+<http://jmri.org/releasenotes/jmri4.9.2.shtml>
 
 Note that JMRI is made available under the GNU General Public License. For more information, please see our copyright and licensing page.
 <http://jmri.org/Copyright.html>
 
 The download links, along with lots of other information which we hope you'll read, can be found on the release note page:
-<http://jmri.org/releasenotes/jmri4.9.1.shtml>
+<http://jmri.org/releasenotes/jmri4.9.2.shtml>
 
 
 - If a production version, update the SF automatic download icon by selecting default in SF.net FRS (3 times)

--- a/xml/schema/types/editors.xsd
+++ b/xml/schema/types/editors.xsd
@@ -1077,6 +1077,7 @@
 
     </xs:complexType>
 
+
     <xs:complexType name="EditorLayoutTurnoutType">
       <xs:annotation>
         <xs:documentation>
@@ -1261,6 +1262,37 @@
       <xs:attribute name="dockY" type="xs:string"></xs:attribute>
       <xs:attribute name="text" type="xs:string"></xs:attribute>
       <xs:attribute name="rosterentry" type="xs:string"></xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="EditorTrainIconType">
+      <xs:annotation>
+        <xs:documentation>
+          Define the XML structure for storing a TrainIcon.
+          Persistence class inherits from jmri.jmrit.display.configurexml.LocoIconXml
+        </xs:documentation>
+        <xs:appinfo>
+          jmri.jmrit.operations.trains.configurexml.TrainIconXml
+          jmri.jmrit.display.configurexml.LocoIconXml
+        </xs:appinfo>
+      </xs:annotation>
+
+      <xs:sequence minOccurs="0" maxOccurs="unbounded">
+        <xs:element name="icon" type="EditorIconType" minOccurs="0" maxOccurs="unbounded"></xs:element>
+        <xs:element name="tooltip" type="xs:string" minOccurs="0" maxOccurs="1"></xs:element>
+      </xs:sequence>
+
+      <xs:attributeGroup ref="EditorCommonAttributesGroup"></xs:attributeGroup>
+      <xs:attributeGroup ref="EditorCommonTextInfoGroup"></xs:attributeGroup>
+      
+      <xs:attribute name="class" type="classType" use="required"></xs:attribute>
+      <xs:attribute name="train" type="xs:string"></xs:attribute>
+      <xs:attribute name="trainName" type="xs:string"></xs:attribute>
+      <xs:attribute name="text" type="xs:string"></xs:attribute>
+      <xs:attribute name="icon" type="yesNoType"></xs:attribute>
+      <xs:attribute name="dockX" type="xs:string"></xs:attribute>
+      <xs:attribute name="dockY" type="xs:string"></xs:attribute>
+      <xs:attribute name="rosterentry" type="xs:string"></xs:attribute>
+
     </xs:complexType>
 
     <xs:complexType name="EditorLocoLabelType">

--- a/xml/schema/types/layouteditor.xsd
+++ b/xml/schema/types/layouteditor.xsd
@@ -41,6 +41,7 @@
 
         <xs:element name="positionablelabel"  type="EditorPositionableLabelType" minOccurs="0" maxOccurs="unbounded" />
         <xs:element name="positionablepoint"  type="EditorPositionablePointType" minOccurs="0" maxOccurs="unbounded" />
+        <xs:element name="trainicon"          type="EditorTrainIconType"         minOccurs="0" maxOccurs="unbounded" />
         <xs:element name="turnouticon"        type="EditorTurnoutIconType"       minOccurs="0" maxOccurs="unbounded" />
         <xs:element name="sensoricon"         type="EditorSensorIconType"        minOccurs="0" maxOccurs="unbounded" />
         <xs:element name="memoryicon"         type="EditorMemoryIconType"        minOccurs="0" maxOccurs="unbounded" />

